modules/openapi/src/software/amazon/smithy/openapi/fromsmithy/protocols/AlloyAbstractRestProtocol.scala

```diff
@@ -426,16 +427,51 @@ abstract class AlloyAbstractRestProtocol[T <: Trait]
     result.asScala
   }
 
+  private def reorganizeExampleTraits(
+      operation: OperationShape,
+      shape: StructureShape
+  ): Shape = {
+    val isErrorShape = shape.hasTrait(classOf[ErrorTrait])
+    val operationOrError =
+      if (isErrorShape) shape
+      else operation
+
+    val allExamples: List[Example] =
+      operation
+        .getTrait(classOf[ExamplesTrait])
+        .asScala
+        .toList
+        .flatMap(_.getExamples.asScala)
+    val allRelevantExamples: List[Example] =
+      // error response so only include examples that are matching to this error shape
+      if (isErrorShape)
+        allExamples
+          .filter(
+            _.getError().asScala.map(_.getShapeId).contains(shape.toShapeId)
+          )
+      // not an error response so no error examples should be included
+      else allExamples.filter(_.getError().isEmpty())
+    val newExamplesTraitBuilder = ExamplesTrait.builder()
+    allRelevantExamples.foreach { ex => newExamplesTraitBuilder.addExample(ex) }
+    val exTrait: Trait = newExamplesTraitBuilder.build()
+    val newShape: Shape =
+      (Shape.shapeToBuilder(operationOrError): AbstractShapeBuilder[_, _])
+        .addTrait(exTrait)
+        .build()
+
+    if (allRelevantExamples.nonEmpty)
+      newShape
+    else operationOrError
```

Comment:
What happens if this is just this?
```suggestion
    newShape
```

Reply:
Good catch, doesn't seem to matter actually